src/main/php/lang/ast/syntax/PHP.class.php

```diff
@@ -1304,13 +1304,40 @@ private function properties($parse, &$body, $meta, $modifiers, $type, $holder) {
       } else {
         $expr= null;
       }
+
       $body[$lookup]= new Property($modifiers, $name, $type, $expr, $annotations, $comment, $line, $holder);
 
-      if (',' === $parse->token->value) {
+      // Check for property hooks
+      if (';' === $parse->token->value) {
+        $parse->forward();
+        return;
+      } else if (',' === $parse->token->value) {
+        $parse->forward();
+        continue;
+      } else if ('{' === $parse->token->value) {
         $parse->forward();
+        do {
+          $hook= $parse->token->value;
+          $parse->forward();
+
+          if ('=>' === $parse->token->value) {
+            $parse->forward();
+            $body[$lookup]->hooks[$hook]= $this->expression($parse, 0);
```

Comment:
If we want to support `get($value)` we will need to store the argument somewhere. Maybe `new Hook($expression, $argument)`?

Reply:
Implemented in a60ac68